source/fe/mapping_q_generic.cc

```diff
@@ -3877,6 +3911,13 @@ compute_mapping_support_points(const typename Triangulation<dim,spacedim>::cell_
         // manifold, otherwise compute them from the points around it
         if (dim != spacedim)
           add_quad_support_points(cell, a);
+        else if (dynamic_cast<const Boundary<dim,dim> *>(&cell->get_manifold()) == NULL)
```

Comment:
This will always be true. For cells in `dim == spacedim`, `cell->get_manifold()` will always return a `Manifold` or a `FlatManifold`.